src/basic/process-util.c

```diff
@@ -1177,7 +1184,11 @@ pid_t clone_with_nested_stack(int (*fn)(void *), int flags, void *userdata) {
         mystack = (uint8_t*) mystack + ps; /* move pointer one page ahead since stacks usually grow backwards */
         mystack = (void*) ALIGN_TO((uintptr_t) mystack, ps); /* align to page size (moving things further ahead) */
 
+#if defined(__ia64__)
+        pid = __clone2(fn, mystack, ps, flags, userdata);
+#else
         pid = clone(fn, mystack, flags, userdata);
+#endif
```

Comment:
hmm, i'd prefer if we'd always use __clone2() if it exists

i.e. make sure HAVE_CLONE2 is defined if glibc exposes the symbol, and then use `#if HAVE_CLONE2` here rather than an arch-specific check.

i.e. arch specific checks suck, feature checks are much nicer. hence, mention in the comments that this is for ia64, but do not code against that.

Reply:
That means this ia64 peculiarity percolates to all other architectures. I don't want to use internal symbols if there's no need to do so. ia64 is almost dead, and once it's removed from the kernel, we can nuke it here too.

Reply:
Moved the prototype to missing_syscall, and left an HAVE_CLONE ifdef here. I am not going to waste a meson check on it, as every new meson check slows down running meson. We know exactly where it's broken here, and with some luck by the end of the year the kernel will have dropped ia64 so we can drop it too.